.github/workflows/check.yml

```diff
@@ -15,10 +15,13 @@ jobs:
         with:
           toolchain: stable
 
-      - name: Start Redis
-        uses: supercharge/redis-github-action@1.4.0
-        with:
-          redis-version: 6
+     - name: Install redis and start redis server
+      run: |
+        sudo apt-get update
+        wget https://github.com/redis/redis/archive/${{ env.REDIS_VERSION }}.tar.gz;
+        tar -xzvf ${{ env.REDIS_VERSION }}.tar.gz;
+        pushd redis-${{ env.REDIS_VERSION }} && BUILD_TLS=yes make && sudo mv src/redis-server src/redis-cli /usr/bin/ && popd;
+        redis-server
```

Comment:
This probably won't work as the processing won't move on to the next step until redis-server exits (which presumably never happens).

Reply:
Will put it in a background so that it keeps running.